neural_speed/application/main_pybind.cpp

```diff
@@ -54,6 +56,241 @@
+                     int batch_size = 1, model_vocab::id pad_token = -1, const std::string& memory_dtype = "auto",
+                     const bool& continuous_batching = false, const int& max_request_num = MODEL_MAX_REQUEST_NUM,
+                     const float& model_scratch_enlarge_scale = 1.0f) {
+  MODEL_ASSERT(params != nullptr);
+#ifdef MODEL_NAME
+  params->model_name = MODEL_NAME;
+#endif
+  params->model_arch = model_name_to_arch::init().find(params->model_name);
+  params->model = model_path;
+  params->n_predict = max_new_tokens;
+  params->n_batch = n_batch;
+  params->n_ctx = ctx_size;
+  params->seed = seed;
+  params->n_threads = threads;
+  params->repeat_penalty = repetition_penalty;
+  params->beam_size = num_beams;
+  params->do_sample = do_sample;
+  params->batch_size = batch_size;
+  params->beam_search = (num_beams > 1 && !do_sample);
+  params->top_k = top_k;
+  params->top_p = top_p;
+  params->temp = temperature;
+  params->n_keep = n_keep;
+  params->n_discard = n_discard;
+  params->shift_roped_k = shift_roped_k;
+  if (memory_dtype == "f32")
+    params->memory_type = KV_MEM_TYPE_F32;
+  else if (memory_dtype == "f16")
+    params->memory_type = KV_MEM_TYPE_F16;
+  else if (memory_dtype == "auto")
+    params->memory_type = KV_MEM_TYPE_AUTO;
+  else
+    fprintf(stderr, "Unexpected memory dtype %s!", memory_dtype.c_str());
+  if (batch_size > 1 && (!continuous_batching || params->model_arch != model_archs::MODEL_GPTJ)) {
+    params->memory_type = KV_MEM_TYPE_F16;  // TODO(Yi & YZT): MHA IN MULTI-BATCH For More Model Archs
+  }
+  params->cont_batching = continuous_batching;
+  params->max_request_num = std::max(batch_size, max_request_num);
+  params->min_new_tokens = min_new_tokens;
+  params->length_penalty = length_penalty;
+  params->do_early_stopping = early_stopping;
+  params->model_scratch_enlarge_scale = model_scratch_enlarge_scale;
+
+  printf(
+      "beam_size: %d, do_sample: %d, top_k: %d, top_p: %f, continuous_batching: %d, max_request_num: %d, "
+      "early_stopping: %d\n",
+      params->beam_size, params->do_sample, params->top_k, params->top_p, params->cont_batching,
+      params->max_request_num, params->do_early_stopping);
+}
+
+class ModelServer {
+ public:
+  ModelServer(const ResponseCallback& response, const std::string& model_path, bool return_prompt, int max_new_tokens,
+              int n_batch, int ctx_size, int seed, int threads, float repetition_penalty, int num_beams, bool do_sample,
+              int top_k, float top_p, float temperature, int min_new_tokens, float length_penalty, bool early_stopping,
+              int n_keep, int n_discard, bool shift_roped_k, int batch_size, model_vocab::id pad_token,
+              const std::string& memory_dtype, const bool& continuous_batching, const int& max_request_num,
+              const float& model_scratch_enlarge_scale, const std::string& policy, const bool& print_log,
+              const std::function<void()>& init_cb)
```

Comment:
usually we don't encourage so much arguments, you can pack them into several structs instead

Reply:
These extra args are for `gpt_params`, as same as the `Model init`. So, pybind `gpt_params` and send it to related python apis? cc @zhenwei-intel